### PR TITLE
Extract stark burner to a separate lib

### PR DESCRIPTION
**Pull request by iossocket**
# Extract stark burner

leverage github insight feature, stark-burner lib: https://github.com/Scaffold-Stark/stark-burner

## Types of change

- [x] Feature
- [ ] Bug
- [ ] Enhancement

### CODE REVIEW

**Comment by metalboyrick** [comment]
@iossocket converting to draft to make sure this PR doesnt get mixed up in the OD Hack PRs

**Review by metalboyrick — COMMENTED** [review]
Where is the burner code stored ser?

**Review by metalboyrick — APPROVED** [review]
code on this side looks good, thanks for the unit tests as well

**Review by metalboyrick — COMMENTED** [review]
might need to do a reconciliation after OD ends

File: packages/nextjs/services/web3/connectors.tsx [hunk 1]
```diff
@@ -23,12 +23,13 @@ function withDisconnectWrapper(connector: InjectedConnector) {
...
 
-  const connectors = [argent(), braavos()];
+  const connectors: InjectedConnector[] = [argent(), braavos()];
 
   if (
     targetNetworks.some((network) => (network.network as string) === "devnet")
   ) {
-    connectors.push(new BurnerConnector());
+    const burnerConnector = new BurnerConnector();
+    connectors.push(burnerConnector as unknown as InjectedConnector);
```
Comment (metalboyrick) [inline packages/nextjs/services/web3/connectors.tsx:32]: just curious what is the reason to refactor it like this?

Comment (iossocket) [reply packages/nextjs/services/web3/connectors.tsx:32]: The root cause of this issue lies in the fact that `InjectedConnector` base class from `@starknet-react/core` has a private `_wallet` property, while `BurnerConnector` as a subclass doesn't explicitly define this property.
In TypeScript, when a class inherits from another class, if the parent class has private properties, the child class must somehow "know" about the existence of these private properties. Due to the scope restrictions of private properties, these properties must come from the same declaration context (same package/module).

Comment (metalboyrick) [reply packages/nextjs/services/web3/connectors.tsx:32]: i see, thanks!

**Review by Nadai2010 — APPROVED** [review]
I have tested it, for my part it is perfect, everything works fine, very clean work, thank you very much.

**Review by 0xquantum3labs — APPROVED** [review]

**Pull request merged**